api/nodes/views.py

```diff
@@ -109,7 +110,11 @@ def perform_destroy(self, instance):
         user = self.request.user
         auth = Auth(user)
         node = self.get_object()
-        node.remove_node(auth=auth)
+        # If there are no children, then allow the node to be removed, otherwise give a 403 error.
+        try:
+            node.remove_node(auth=auth)
+        except NodeStateError:
+            raise PermissionDenied()
```

Comment:
I'm not sure this is the correct response to send if a `NodeStateError` is raised. Take a look at the `Node#remove_node` method and see what conditions raise `NodeStateError`. It's also possible for a `PermissionsError` to be raised, which should also be handled.